tests/foreman/cli/test_docker.py

```diff
@@ -63,3 +75,874 @@ def test_bugzilla_1190122(self):
+        content view and publish it. Create an activation key
+        and associate it with the Docker content view. Then remove
+        this content view from the activation key.
+
+        @Assert: Docker-based content view can be added and then removed
+        from the activation key.
+
+        @Feature: Docker
+
+        @Status: Manual
+
+        """
+
+    @stubbed()
+    @run_only_on('sat')
+    def test_add_docker_repo_composite_view_to_activation_key(self):
+        """@Test:Add Docker-type repository to a non-composite
+        content view and publish it. Then add this content view to a composite
+        content view and publish it. Create an activation key and associate it
+        with the composite Docker content view.
+
+        @Assert: Docker-based content view can be added to activation key
+
+        @Feature: Docker
+
+        @Status: Manual
+
+        """
+
+    @stubbed()
+    @run_only_on('sat')
+    def test_remove_docker_repo_composite_view_to_activation_key(self):
+        """@Test:Add Docker-type repository to a non-composite
+        content view and publish it. Then add this content view to a composite
+        content view and publish it. Create an activation key and associate it
+        with the composite Docker content view. Then, remove the composite
+        content view from the activation key.
+
+        @Assert: Docker-based composite content view can be added and then
+        removed from the activation key.
+
+        @Feature: Docker
+
+        @Status: Manual
+
+        """
+
+
+@ddt
+class DockerClientTestCase(CLITestCase):
+    """Tests specific to using ``Docker`` as a client to pull Docker images from
+    a Satellite 6 instance."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an organization and product which can be re-used in tests."""
+        super(DockerClientTestCase, cls).setUpClass()
+        cls.org_id = entities.Organization().create_json()['id']
+
+    @stubbed()
+    @run_only_on('sat')
+    def test_docker_client_pull_image(self):
+        """@Test: A Docker-enabled client can use ``docker pull`` to pull a
+        Docker image off a Satellite 6 instance.
+
+        @Steps:
+            1.  Publish and promote content view with Docker content
+            2.  Register Docker-enabled client against Satellite 6.
+
+        @Assert: Client can pull Docker images from server and run it.
+
+        @Status: Manual
+
+        """
+
+    @stubbed()
+    @run_only_on('sat')
+    def test_docker_client_upload_image(self):
+        """@Test: A Docker-enabled client can create a new ``Dockerfile``
+        pointing to an existing Docker image from a Satellite 6 and modify it.
+        Then, using ``docker build`` generate a new image which can then be
+        uploaded back onto the Satellite 6 as a new repository.
+
+        @Steps:
+            1.  Publish and promote content view with Docker content
+            2.  Register Docker-enabled client against Satellite 6.
```

Comment:
@Steps:
    
    1. Publish and promote content view with Docker content.
    2. Register Docker-enabled client against Satellite 6.